### PR TITLE
fix(frontend/shared): add confirm password field on sign up page bb-144

### PR DESCRIPTION
**Pull request by marharita08**
## Summary
Added confirm password field on sign up page.
## Added
- UserSignUpFormDto type.
- ConfirmPassword enum.
## Updated
- validation schema for sign up: added missing rules for password.
- useAppForm hook to export setError function.
- sign up component: added Confirm password input and function to handle passwords mismatch.
## Screenshots
image

image

### CODE REVIEW

File: apps/frontend/src/pages/auth/components/sign-up-form/libs/constants.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { type UserSignUpRequestDto } from "~/modules/users/users.js";
 
 const DEFAULT_SIGN_UP_PAYLOAD: UserSignUpRequestDto = {
+	confirmPassword: "",
```
Comment (ihorLysak) [inline apps/frontend/src/pages/auth/components/sign-up-form/libs/constants.ts:4]: Do we need to send confirmPassword to server? I feel like this validation can be done on client

Comment (marharita08) [reply apps/frontend/src/pages/auth/components/sign-up-form/libs/constants.ts:4]: Yeah this field is not needed on server. But in useAppForm hook specified type UserSignUpRequestDto, so I can't add field that is missing in this type. And when I delete property confirmPassword from UserSignUpRequestDto I faced an error with confirm password input.

File: packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 type UserSignUpRequestValidationDto = {
+	confirmPassword: z.ZodString;
 	email: z.ZodString;
 	name: z.ZodString;
 	password: z.ZodString;
 };
 
 const userSignUp = z
 	.object<UserSignUpRequestValidationDto>({
+		confirmPassword: z.string().trim(),
```
Comment (fshabanov) [inline packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts:15]: Shouldn't be a part of validation and shouldn't be sent to Backend. You should handle it in sign up onSubmit function

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

File: packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts [hunk 3]
```diff
@@ -22,7 +22,18 @@ const userSignUp = z
...
+			.trim()
+			.min(UserValidationRule.NON_EMPTY_STRING_MIN_LENGTH, {
+				message: UserValidationMessage.PASSWORD_REQUIRE,
+			})
+			.max(UserValidationRule.PASSWORD_MAX_LENGTH, {
+				message: UserValidationMessage.PASSWORD_MAX_LENGTH,
+			})
+			.min(UserValidationRule.PASSWORD_MIN_LENGTH, {
+				message: UserValidationMessage.PASSWORD_MIN_LENGTH,
+			}),
```
Comment (fshabanov) [inline packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts:36]: There are 2 `min` values, should be only 1

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/auth/components/sign-up-form/libs/constants.ts [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
-const DEFAULT_SIGN_UP_PAYLOAD: UserSignUpRequestDto = {
+const DEFAULT_SIGN_UP_PAYLOAD: UserSignUpFormDto = {
+	confirmPassword: "",
 	email: "",
 	name: "",
 	password: "",
 };
+const CONFIRM_PASSWORD_KEY = "confirmPassword";
+const CONFIRM_PASSWORD_ERROR_MESSAGE = "Passwords don't match";
+const CONFIRM_PASSWORD_ERROR_TYPE = "custom";
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/components/sign-up-form/libs/constants.ts:11]: B29

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by revenets — APPROVED** [review]

File: apps/frontend/src/pages/auth/components/sign-up-form/libs/enums.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const ConfirmPassword = {
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/components/sign-up-form/libs/enums.ts:1]: Name is not clear

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx [hunk 6]
```diff
@@ -1,28 +1,42 @@
 import { Button, Input } from "~/libs/components/components.js";
 import { useAppForm, useCallback } from "~/libs/hooks/hooks.js";
 import {
+	type UserSignUpFormDto,
 	type UserSignUpRequestDto,
 	userSignUpValidationSchema,
 } from "~/modules/users/users.js";
 
 import { DEFAULT_SIGN_UP_PAYLOAD } from "./libs/constants.js";
+import { CustomValidation } from "./libs/enums.js";
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx:10]:
```suggestion
import { CustomValidation } from "./libs/enums/enums.js";
```

Comment (fshabanov) [inline apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx:10]:
```suggestion
import { SignUpCustomValidation } from "./libs/enums.js";
```

File: packages/shared/src/modules/users/libs/enums/custom-validation.enum.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const CustomValidation = {
```
Comment (fshabanov) [inline packages/shared/src/modules/users/libs/enums/custom-validation.enum.ts:1]: Move to auth folder

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by WilhelmWesser — APPROVED** [review]

**Pull request merged**